WIP/PlayerSession.py

```diff
@@ -0,0 +1,147 @@
+from ServerConnection import ServerConnection
+from enum import IntEnum
+
+resultDict = dict[str, any]  # result dictionary type
+
+
+class Result(IntEnum):
+    """
+    Enumerator of all possible result codes given in the docs.
+    """
+    OKAY = 0
+    BAD_COMMAND = 1
+    ACCESS_DENIED = 2
+    INAPPROPRIATE_GAME_STATE = 3
+    TIMEOUT = 4
+    INTERNAL_SERVER_ERROR = 500
+
+    def __eq__(self, other):
+        if isinstance(other, Result):
+            return self.value == other.value
+
+        return self.value == other  # if it's int
+
+
+class PlayerSession:
+    __slots__ = ("name", "connection")  # class members
+
+    def __init__(self, name):
+        self.name = name
+
+    def __enter__(self):
+        self.connection = ServerConnection()
+        return self
+
+    @staticmethod
+    def __handleResult(result):
+        """
+        Handles error codes and returns data if request was successful
+        :param result: result dict from action request with "resultCode" and "data" keys
+        :return: result["data"] if the result was okay,
+        None if it was TIMEOUT error.
+        Exception is raised for other error types.
+        """
+        code = result["resultCode"]
+        if code == Result.BAD_COMMAND or code == Result.INAPPROPRIATE_GAME_STATE or code == Result.ACCESS_DENIED or code == Result.INTERNAL_SERVER_ERROR:
+            raise Exception("ERROR: " + (Result(code)).name + "\n" + result["data"]["error_message"])
+        elif code == Result.TIMEOUT:
+            return None  # Action should be requested again
+        return result["data"]
+
+    def login(self) -> int:
+        """
+        Logs the player to the game server.
+        :return: player id if login was successful, -1 otherwise
+        """
+        data = dict()
+        data["name"] = self.name
+        data["game"] = "testtesttest"
```

Comment:
I don't quite understand. I've already seen similar code. Why do you duplicated it?